app/generic/definitions.py

```diff
@@ -248,11 +257,69 @@ async def create_credential_definition(
             raise CloudApiException(
                 "Unable to construct credential definition id from signature response"
             ) from e
-
     else:
-        await stop_listener()
         credential_definition_id = result.credential_definition_id
 
+    if credential_definition.support_revocation:
+        try:
+            # Create a revocation registry and publish it on the ledger
+            revoc_reg_creation_result = await create_revocation_registry(
+                controller=aries_controller,
+                credential_definition_id=credential_definition_id,
+                max_cred_num=credential_definition.revocation_registry_size,
+            )
+            await aries_controller.revocation.update_registry(
+                rev_reg_id=revoc_reg_creation_result.revoc_reg_id,
+                body=RevRegUpdateTailsFileUri(
+                    tails_public_uri=f"{ACAPY_TAILS_SERVER_BASE_URL}/{revoc_reg_creation_result.revoc_reg_id}"
+                ),
+            )
+            endorser_connection = await aries_controller.connection.get_connections(
+                alias=ACAPY_ENDORSER_ALIAS
+            )
+            # NOTE: Special case - the endorser registers a cred def itself that
+            # supports revocation so there is no endorser connection.
+            # Otherwise onboarding should have created an endorser connection
+            # for tenants so this fails correctly
+            has_connections = len(endorser_connection.results) > 0
+            await publish_revocation_registry_on_ledger(
+                controller=aries_controller,
+                revocation_registry_id=revoc_reg_creation_result.revoc_reg_id,
+                connection_id=endorser_connection.results[0].connection_id
+                if has_connections
+                else None,
+                create_transaction_for_endorser=has_connections,
+            )
+            if has_connections:
+                async with get_governance_controller() as endorser_controller:
+                    wait_for_event_with_timeout, stop_listener = await start_listener(
+                        topic="endorsements", wallet_id="admin"
+                    )
+                    try:
+                        txn_record = await wait_for_event_with_timeout(
+                            filter_map={
+                                "state": "request-received",
+                            },
+                            timeout=30,
+                        )
+                    except TimeoutError:
+                        raise CloudApiException(
+                            "Failed to retrieve transaction record for endorser", 500
+                        )
+                    finally:
+                        await stop_listener()
+
```

Comment:
starting the listener after the actions that triggers the event can possibly lead to race conditions